pkgs/applications/virtualization/xen/0004-fix-oxenstored-systemd-notification-race.patch

```diff
@@ -0,0 +1,15 @@
+diff -ru xen/tools/hotplug/Linux/launch-xenstore.in xen-delay1/tools/hotplug/Linux/launch-xenstore.in
+--- xen/tools/hotplug/Linux/launch-xenstore.in	2017-05-27 10:19:32.000000000 +0200
++++ xen-delay1/tools/hotplug/Linux/launch-xenstore.in	2017-05-29 23:33:53.000000000 +0200
+@@ -66,6 +66,11 @@
+ 
+ 	systemd-notify --booted 2>/dev/null || timeout_xenstore $XENSTORED || exit 1
+ 
++	# We wait for 10 seconds here as otherwise the sd_notify() notification
++	# from the oxenstored process might not be properly registered by systemd
++	# if this process quits before the notification is sent.
++	sleep 10
```

Comment:
The sleep 10 is fragile, as pointed out by others. If it cannot be done in any other way, leave it like this, but if it can be fixed, fix it. What happened to the bug report to Xen?